### ISSUE TITLE
Improve error message when energy plugin is not activated

### ISSUE THREAD

**Issue by rafaelfsilva** [wrench-project/wrench#89]
Need to properly handle the exception to override SimGrid's exception. Instead of the message below, need to print a message to run the simulator with `--activate-energy`.

```
[0.000000][master:energy_meter_19(11)] The Energy plugin is not active. Please call sg_host_energy_plugin_init() during initialization.
```

**Comment by henricasanova** [comment]
Unfortunately, SImGrid simply calls "abort()", which cannot be caught. This is because SimGrid still supports C and C++ I think.  So there is nothing we can do right now.  This is related to the "Cleaner Exceptions" problem (Issue #7).  I'll ping the SimGrid guys about this again.

**Comment by frs69wq** [comment]
I’m pushing @mquinson to have SimGrid to spill out more exceptions (and have a nice hierarchy too). Maybe if you ask for it too, you might convince him. Meanwhile, if you tell me what you’re missing, I can have a look.

**Comment by henricasanova** [comment]
I actually sent e-mail to Martin.  One thing that would be nice would be if SG threw something whenever it does an xbt_assert() for the Energy Plugin. For instance, in plugins/host_energy.cpp.  More generally, xbt_assert() ends up doing an abort(), when instead it should throw something. It could throw some FatalError exception, and then we  have a type (e.g., ENERGY_PLUGIN_NOT_ACTIVATED would be a type)...

**Comment by frs69wq** [comment]
I think we have all the plumbing to throw an exception where you need it. there may be this “we still have to support C users” issue, but there has to be way to please everyone. I’ll see what I can do. 
do not hesitate to send such requests either on simgrid-user or simgrid-dev, so that the other contibutora have a chance to help you.

**Comment by mquinson** [comment]
I do agree with @frs69wq. SimGrid is closer than ever to the Exceptional Conversion. I first tried maybe 6 months ago, and a lot of things caught fire and started exploding all around the place. Things cooled down very recently, and @frs69wq is back on track, cleaning up the activity code.

We should try again to push the exceptions and see whether our codebase holds its ground, this time.

**Comment by henricasanova** [comment]
Tried this again, but SimGrid doesn't throw for this yet, and just aborts. so we can't catch it. Let's wait for the next release :)

**Comment by mquinson** [comment]
I think that this is fixed in https://github.com/simgrid/simgrid/commit/d9fb9c3b94a08a8a49509e49e3dc847f6c215b99 : we now throw a std::logic_error if the plugin is not inited before use. 
Please tell me how it work for you.

**Comment by rafaelfsilva** [comment]
This issue has been fixed on commit b093aa2c9ce576cd3626f5d33b774fb9d94822be.
The current exception prints:
```
Uncaught exception std::runtime_error: S4U_Simulation::getEnergyConsumedByHost(): Was not able to get the energy consumed by the host. Make sure the energy plugin is enabled (--activate-energy)
```
